src/commands/infra/generate.ts

```diff
@@ -0,0 +1,190 @@
+        );
+        logger.error(err);
+      }
+    });
+};
+
+/**
+ * Checks if definition.json is present locally to provided project path
+ *
+ * @param projectPath Path to the definition.json file
+ */
+export const validateDefinition = async (
+  projectPath: string
+): Promise<boolean> => {
+  try {
+    // If templates folder does not exist, create cache templates directory
+    if (!fs.existsSync(spkTemplatesPath)) {
+      fs.mkdirSync(spkTemplatesPath);
+    }
+    if (!fs.existsSync(path.join(projectPath, "definition.json"))) {
+      logger.error(
+        `Provided project path for generate is invalid or definition.json can not be found: ${projectPath}`
+      );
+      return false;
+    }
+    logger.info(
+      `Project folder found. Attempting to generate definition.json file.`
+    );
+  } catch (_) {
+    logger.error(`Unable to validate project folder path.`);
+    return false;
+  }
+  return true;
+};
+
+/**
+ * Checks if working definition.json is present in the provided project path with validated source & version
+ *
+ * @param projectPath Path to the definition.json file
+ */
+export const validateTemplateSource = async (
+  projectPath: string
+): Promise<string[]> => {
+  try {
+    const rootDef = path.join(projectPath, "definition.json");
+    const data: string = fs.readFileSync(rootDef, "utf8");
+    const definitionJSON = JSON.parse(data);
+    // TO DO : Check for malformed JSON
+    if (!(definitionJSON.template && definitionJSON.source)) {
+      logger.info(
+        `The definition.json file is invalid. There is a missing field for the definition file's sources. Template: ${definitionJSON.template} source: ${definitionJSON.source} version: ${definitionJSON.version}`
+      );
+      return [];
+    }
+    logger.info(
+      `Checking for locally stored template: ${definitionJSON.template} from remote repository: ${definitionJSON.source} at version: ${definitionJSON.version}`
+    );
+    const sources = [
+      definitionJSON.source,
+      definitionJSON.template,
+      definitionJSON.version
+    ];
+    return sources;
+  } catch (_) {
+    logger.error(
+      `Unable to validate project folder definition.json file. Is it malformed?`
+    );
+    return [];
+  }
+  return [];
+};
+
+/**
+ * Checks if provided source, template and version are valid. TODO/ Private Repo, PAT, ssh-key agent
+ *
+ * @param projectPath Path to the definition.json file
+ */
+export const validateRemoteSource = async (
+  definitionJSON: string[]
+): Promise<boolean> => {
+  try {
+    const [source, template, version] = definitionJSON;
+    // Converting source name to storable folder name
+    const httpReg = /^(.*?)\.com/;
+    const punctuationReg = /[^\w\s]/g;
+    let sourceFolder = source.replace(httpReg, "");
+    sourceFolder = sourceFolder.replace(punctuationReg, "_").toLowerCase();
+    const sourcePath = path.join(spkTemplatesPath, sourceFolder);
```

Comment:
Should the presence of spkTemplatesPath be checked for here rather than in the prior function?

Reply:
`fs` during testing was not waiting for the return of checking for the folder and bypassing the check for the template. I placed it into another function to force the promise to await in `fs`. Open to suggestions